src/daemon/daemon.c

```diff
@@ -1360,16 +1360,17 @@ static gboolean notify_daemon_notify_handler(NotifyDaemonNotifications *object,
 	 */
 
 
-	if (g_variant_lookup(hints, "window-xid", "v", &data))
+	if (g_variant_lookup(hints, "window-xid", "@u", &data))
 	{
 		window_xid = (Window) g_variant_get_uint32 (data);
+		g_variant_unref(data);
 	}
 	/* deal with x, and y hints */
-	else if (g_variant_lookup(hints, "x", "v", &data))
+	else if (g_variant_lookup(hints, "x", "i", &x))
 	{
 		x = g_variant_get_int32 (data);
```

Comment:
This `g_variant_get_int32` call needs to be removed. The `g_variant_lookup` call before already assigns the value to `x`. `data` is going to be either uninitialized or freed at this point.